src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -124,7 +124,7 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             {
                 rightParts = ((BoundTupleExpression)right).Arguments;
             }
-            else if (right.Kind == BoundKind.Conversion)
+            else if (right.Kind == BoundKind.Conversion && ((BoundConversion)right).Operand.Kind == BoundKind.ConvertedTupleLiteral)
```

Comment:
:memo: Even though there is a conversion, the operand may not be a tuple literal. In such case, we should fall through to the next case (which handles tuple types).